sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -105,193 +103,39 @@ public void setup() throws Exception {
-                        payload = resultSet.getInt(i);
-                        break;
-                      case Types.BLOB:
-                        payload = resultSet.getBlob(i);
-                        break;
-                      case Types.BOOLEAN:
-                        payload = resultSet.getBoolean(i);
-                        break;
-                      case Types.CHAR:
-                        payload = resultSet.getString(i);
-                        break;
-                      case Types.CLOB:
-                        payload = resultSet.getClob(i);
-                        break;
-                      case Types.DATE:
-                        payload = resultSet.getDate(i);
-                        break;
-                      case Types.DECIMAL:
-                        payload = resultSet.getBigDecimal(i);
-                        break;
-                      case Types.DOUBLE:
-                        payload = resultSet.getDouble(i);
-                        break;
-                      case Types.FLOAT:
-                        payload = resultSet.getFloat(i);
-                        break;
-                      case Types.INTEGER:
-                        payload = resultSet.getInt(i);
-                        break;
-                      case Types.LONGNVARCHAR:
-                        payload = resultSet.getString(i);
-                        break;
-                      case Types.LONGVARCHAR:
-                        payload = resultSet.getString(i);
-                        break;
-                      case Types.NCHAR:
-                        payload = resultSet.getNString(i);
-                        break;
-                      case Types.NCLOB:
-                        payload = resultSet.getNClob(i);
-                        break;
-                      case Types.SMALLINT:
-                        payload = resultSet.getInt(i);
-                        break;
-                      case Types.TIME:
-                        payload = resultSet.getTime(i);
-                        break;
-                      case Types.TIMESTAMP:
-                        payload = resultSet.getTimestamp(i);
-                        break;
-                      case Types.TINYINT:
-                        payload = resultSet.getInt(i);
-                        break;
-                      case Types.VARCHAR:
-                        payload = resultSet.getString(i);
-                        break;
-                      default:
-                        payload = resultSet.getObject(i);
-                        break;
-                    }
-                    record.getColumnValues()[i - 1] = payload;
-                  }
+                  KV<Integer, String> kv =
+                      KV.of(resultSet.getInt("id"), resultSet.getString("name"));
+                  return kv;
                 } catch (Exception e) {
-                  LOGGER.error("Can't map row", e);
+                  return null;
```

Comment:
This code says "if can't parse record, silently drop it and return null instead" - this is silent data loss and it's a poor example to show to users, especially given that the documentation of JdbcIO delegates to the test for a complete example. As a user, I would be very surprised to find nulls in my PCollection with no indication of how they got there. I'd much prefer if the read failed, than if it silently produced nulls.

Please remove the error handling and let it fail. I think this is a general recommendation that I've made several times in these PRs: the runner does error handling for you. Unless you are dealing with a rare case where the error is transient/recoverable AND your logic does a better job of recovering from it than the runner would have (by retrying your bundle), you should not have error handling code at all (I can think of only one case: retrying individual RPCs in case of transient RPC errors, but usually various client libraries do that for you).

And most importantly, it is absolutely never OK to have error handling code that masks a data-loss error (e.g. failure to write data or failure to parse data).